Framework/Algorithms/test/MaxEnt/MaxentTransformMultiFourierTest.h

```diff
@@ -0,0 +1,232 @@
+#include <boost/shared_ptr.hpp>
+#include <boost/make_shared.hpp>
+#include <cmath>
+
+using Mantid::Algorithms::MaxentSpaceComplex;
+using Mantid::Algorithms::MaxentSpaceReal;
+using Mantid::Algorithms::MaxentTransformFourier;
+using Mantid::Algorithms::MaxentTransformMultiFourier;
+using Mantid::Algorithms::MaxentSpaceComplex_sptr;
+using Mantid::Algorithms::MaxentSpace_sptr;
+
+class MaxentTransformMultiFourierTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static MaxentTransformMultiFourierTest *createSuite() {
+    return new MaxentTransformMultiFourierTest();
+  }
+  static void destroySuite(MaxentTransformMultiFourierTest *suite) {
+    delete suite;
+  }
+
+  void test_complex_data_to_real_image_against_fourier() {
+
+    MaxentSpaceComplex_sptr dataSpaceMF =
+        boost::make_shared<MaxentSpaceComplex>();
+    MaxentSpace_sptr imageSpaceMF = boost::make_shared<MaxentSpaceReal>();
+    MaxentTransformMultiFourier transformMF(dataSpaceMF, imageSpaceMF, 3);
+    MaxentSpace_sptr dataSpaceF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentSpace_sptr imageSpaceF = boost::make_shared<MaxentSpaceReal>();
+    MaxentTransformFourier transformF(dataSpaceF, imageSpaceF);
+
+    // Three square waves that add up to a saw tooth wave
+    std::vector<double> realDataMF = {1, -1, 1,  -1, 1,  -1, 1,  -1,
+                                      2, 2,  -2, -2, 2,  2,  -2, -2,
+                                      4, 4,  4,  4,  -4, -4, -4, -4};
+    std::vector<double> realDataF = {7, 5, 3, 1, -1, -3, -5, -7};
+
+    // Perform the transformation
+    auto result = transformMF.dataToImage(realDataMF);
+    // Perform Fourier transformation for comparison
+    auto resultF = transformF.dataToImage(realDataF);
+
+    // Check both results are equal
+    TS_ASSERT_EQUALS(result.size(), resultF.size());
+    if (result.size() == resultF.size()) {
+      for (size_t i = 0; i < result.size(); i++) {
+        TS_ASSERT_DELTA(result[i], resultF[i], 1e-4);
+      }
+    }
+  }
+
+  void test_complex_data_to_complex_image_against_fourier() {
+
+    MaxentSpaceComplex_sptr dataSpaceMF =
+        boost::make_shared<MaxentSpaceComplex>();
+    MaxentSpace_sptr imageSpaceMF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentTransformMultiFourier transformMF(dataSpaceMF, imageSpaceMF, 3);
+    MaxentSpace_sptr dataSpaceF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentSpace_sptr imageSpaceF = boost::make_shared<MaxentSpaceComplex>();
+    MaxentTransformFourier transformF(dataSpaceF, imageSpaceF);
+
+    std::vector<double> realDataMF = {1, -1, 1,  -1, 1,  -1, 1,  -1,
+                                      2, 2,  -2, -2, 2,  2,  -2, -2,
+                                      4, 4,  4,  4,  -4, -4, -4, -4};
+    std::vector<double> realDataF = {7, 5, 3, 1, -1, -3, -5, -7};
+
```

Comment:
sorry, why should these be the same?

Reply:
Please explain in more detail.

Reply:
Perhaps, that is because I comparing with the effects of the fourrier transform. I'll have look into this.
I have not tested the Fourrier transform here, because it is tested by MaxentTransformFourier.